[WIP] Add support for other SIPS2 APIs

Jydett opened the pull request:
Hi, this WIP PR adds support to others SIPS2 API (only Office -> wallet for now)

 - This include a refactor to extract the common parts
 - The module common-payment-sdk is renamed common-sdk
 - A new module wallet-sdk is created

TODO
- [X] Refactor code base
- [X] Payement tests workings with refactor
- [x] Implements wallets methods (getWalletData & deletePaymentMean)
- [x] TU
- [x]  Javadoc

Feedback appreciated !

Commit:
[WIP] Add Wallet API and refactor common code
 - factorize common code in sdk-common
 - remove sdk-payment common

No javadoc yet

Commit:
Merge branch 'master' into develop

Commit:
Merge branch 'develop' of https://github.com/Jydett/sips-sdk-java into worldline_develop

 Conflicts:
	payment-sdk/build.gradle

Commit:
[WIP] rebased to 1.4.4-SNAPSHOT

No javadoc yet

Commit:
[WIP] Fix serialisation & seal computation
 - Seal computation now sort all fields of a class hierarchy instead of sorting on each class
 - Fix broken seal calculator tests

Commit:
[WIP] Add all payment mean brand

Commit:
[WIP] Add Delete Payment Mean Request

CrlH commented:
Great work so far, @Jydett!

Couple remarks up until now

- SIPS 1.0 is EOL and only ever sold in FR. I’d propose not to use any sips 2.0 references, as this will cause unneeded confusion

- ING Home’Pay is discontinued by ING and will shortly be removed from SIPS

Jydett committed:
[WIP] javadoc & formatting

Jydett committed:
[WIP] formatting

Jydett commented:
Ok @CrlH thx for the feedback.
I'll remove the SIPS 2.0 references and ING HomePay.

A couple a unit tests and documentations and this should be ready to review :D

On a side note are you interested to add github actions to this project ? (this could be done in another PR if you are interested)

Jydett committed:
[WIP] Delete ING' HOMEPAY & remove ref to sips 2

Commit:
[WIP] Rename common sdk

 - fix javadoc
 - fix exception bad imports
 - remove unused code
 - add few tests
 - accept unknown response code

Commit:
[WIP] Remove todos

Commit:
[WIP] Rename common sdk

 - fix javadoc
 - fix exception bad imports
 - remove unused code
 - add few tests
 - accept unknown response code

Commit:
Merge remote-tracking branch 'origin/develop' into develop

Jydett commented:
This si worth noting that this could be a major version since some imports and names changed

Commit:
Update README.md

Jydett commented:
I've updated the readme **without** changing the version number since I'm not sure how you want to bump the lib version with  this PR

Commit:
Fix seal computation for response requests
 - YearMonth is correctly parsed in seal computation
 - Add s10TransactionReference for Sips1 retro-compatibility
 - Fix payment response parameters type
 - Offesetted type zone are now correctly parsed

Commit:
Add support for unknown payement brand and mean

Commit:
Add support for unknown payment brand and mean
Fix payment test
